Streetcode/Streetcode.DAL/Repositories/Interfaces/Base/IRepositoryBase.cs

```diff
@@ -1,21 +1,21 @@
 
 namespace StreetCode.DAL.Repositories.Interfaces.Base;
 
-public interface IRepositoryBase
+public interface IRepositoryBase<T>
 {
 
-    public void Create();
+     void Create();
 
-    public void CreateAsync();
+     Task CreateAsync();
 
-    public void GetById();
+     void GetById();
 
-    public void GetAll();
+     IQueryable<T> GetAll();
```

Comment:
When working with IQueryable interface, we are posing a "question", applying all necessary filters right off the bat.
Hence the method's name should be "FindAll" and not "GetAll".